src/data/battler-tags/type-immune-tag.ts

```diff
@@ -0,0 +1,24 @@
+import { BattlerTag } from "#app/data/battler-tags/battler-tag";
+import { BattlerTagLapseType } from "#enums/battler-tag-lapse-type";
+import type { BattlerTagType } from "#enums/battler-tag-type";
+import type { ElementalType } from "#enums/elemental-type";
+import type { MoveId } from "#enums/move-id";
+
+export class TypeImmuneTag extends BattlerTag {
```

Comment:
Is this tag needed? The only thing that uses this is the FloatingTag

Reply:
if kept, it could be abstract
```suggestion
export abstract class TypeImmuneTag extends BattlerTag {
```